### ISSUE TITLE
Question About Application Listening Address

### ISSUE THREAD

**Issue by emnul** [dreamsofcode-io/guestbook#1]
Hi I'm a bit confused about how this application works and was hoping you could help me better understand what I might be missing here. Specifically, I'm confused about how this application is able to serve requests at all, since as far as I understand it, the application should never receive requests on the loopback address from Traefik.

Traefik routes requests to the internal network IP/port of the container in the range of 172.... Additionally, the loopback address is only acceessible within the container itself. It's not accessible by Traefik or other containers, so Traefik will not be able to find a binding to the guestbook container and fall back to the internal IP/Port anyways.

I'm curious how it's possible that this application can serve requests while listening to `127.0.0.1:8080`? I'm building a [personal blog](https://github.com/emnul/blog-server/tree/main) and found that my application threw the following error `502 Bad Gateway error="dial tcp 172.18.0.3:8080: connect: connection refused"` when configured to listen to `127.0.0.1:8080` for requests, and this makes sense. What am I missing here?

Thanks for creating the guide on your YouTube channel! It's been immensely helpful.

**Comment by elliottminns** [comment]
Hey! You're correct. It won't work when binding to 127.0.0.1

That's a mistake on my part, I've actually been developing this project for another video and so I commited that change by accident to the main branch.

Instead it's meant to bind to 0.0.0.0:8080 or just :8080 which will allow external traffic to the container.


I've reverted the change now so it should work if you check the most recent commit!
